### DISCUSSION TITLE
Sticky navbar

### DISCUSSION THREAD

**Discussion by LITUATUI** [adityatelange/hugo-PaperMod#1151] [Ideas]
Please add the option to make the navbar sticky, or make it default.

Thanks!

**Comment by masonwilde** [comment]
~~You can add `style="position:fixed;top:0px;width:100%"` to the nav element in either layouts/partials/header.html or in assets/css/header.css.~~

The above doesn't work well because it causes the next element to clip over the header and block the clickable parts.

I got around that, but it's not pretty.

In `layouts/partials/header.html`:
Add `style="position:fixed;top:0px;width:100%"` to the header, then add a `<div id="header-offset"></div>` after the header element.

Finally, add
```
<script>
    let header = document.getElementsByClassName("header")[0];
    let headerOffset = document.getElementById("header-offset");
    headerOffset.style.height = header.firstChild.offsetHeight + "px";
</script>
```
After the div. This gets the height of the header (but probably breaks if there is more than one child and would need some sort of loop logic to add up all children while accounting for inline ones...) and makes the header-offset div match it.

Here's a full [header.html](https://gist.github.com/masonwilde/296d2537c736bf159585f57f1a0e849c)

Disclaimer: I'm pretty new to web stuff in general, so it's highly likely there is a better way to do this...
